core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,705 @@
+    // The sync should fail
+    mUfs.mThrowIOException = false;
+    syncScheme = new LockingScheme(new AlluxioURI(TEST_DIR_PREFIX + dirNum + TEST_FILE_PREFIX),
+        InodeTree.LockPattern.READ,
+        true); //shouldSync
+    syncStream = new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+        DescendantType.NONE, options,
+        false, //isGetFileInfo
+        false, //forceSync
+        false, // loadOnly
+        false); // loadAlways
+    syncStream.sync();
+
+    assertEquals(streamCount + 1, streamCountCounter.getCount());
+    streamCount += 1;
+    assertEquals(succeededStreams + 0, succeededStreamCounter.getCount());
+    succeededStreams += 0;
+    assertEquals(succeededPaths + 0, succeededPathCounter.getCount());
+    succeededPaths += 0;
+    assertEquals(failedStreams + 1, failedStreamCounter.getCount());
+    failedStreams += 1;
+    assertEquals(failedPaths + 1, failedPathCounter.getCount());
+    failedPaths += 1;
+    assertEquals(noChangedPaths + 0, noChangedPathsCounter.getCount());
+    noChangedPaths += 0;
+    assertEquals(skippedStreams + 0, skippedStreamCounter.getCount());
+    skippedStreams += 0;
+  }
+
+  @Test
+  public void metadataPrefetchMetrics() throws Exception {
+    final Counter prefetchOpsCountCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_OPS_COUNT;
+    final Counter succeededPrefetchCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_SUCCESS;
+    final Counter failedPrefetchCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_FAIL;
+    final Counter canceledPrefetchCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_CANCEL;
+    final Counter prefetchPathsCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_PATHS;
+    final Counter prefetchRetriesCounter =
+        DefaultFileSystemMaster.Metrics.METADATA_SYNC_PREFETCH_RETRIES;
+
+    int previousPrefetchOpsCount = 0;
+    int previousSucceededPrefetches = 0;
+    int previousFailedPrefetches = 0;
+    int previousCancelPrefetches = 0;
+    int previousPrefetchPaths = 0;
+
+    UfsStatusCache ufsStatusCache = new UfsStatusCache(mUfsStateCacheExecutorService,
+        mFileSystemMaster.getAbsentPathCache(), UfsAbsentPathCache.ALWAYS);
+    MountTable mountTable = mFileSystemMaster.getMountTable();
+
+    String dir0 = TEST_DIR_PREFIX + "0";
+    createUfsDir(dir0);
+    createUfsFile(dir0 + TEST_FILE_PREFIX + "0").close();
+    createUfsFile(dir0 + TEST_FILE_PREFIX + "1").close();
+    createUfsFile(dir0 + TEST_FILE_PREFIX + "2").close();
+    String dir1 = TEST_DIR_PREFIX + "1";
+    createUfsFile(dir1 + TEST_FILE_PREFIX + "0").close();
+    createUfsFile(dir1 + TEST_FILE_PREFIX + "1").close();
+    createUfsFile(dir1 + TEST_FILE_PREFIX + "2").close();
+    String dir2 = TEST_DIR_PREFIX + "2";
+
+    //The path is existent in UFS
+    //The prefetch should succeed
```

Comment:
check all your other comments
```suggestion
    // The path is existent in UFS
    // The prefetch should succeed
```